### DISCUSSION TITLE
Approaching merge SwiftyRemoteConfig into firebase-ios-sdk as RemoteConfigSwift.

### DISCUSSION THREAD

**Discussion by fumito-ito** [firebase/firebase-ios-sdk#9108] [Pitches]
Hi, I got following message on twitter for my library, [SwiftyRemoteConfig](https://github.com/fumito-ito/SwiftyRemoteConfig). 

> Looks promising. We'll be happy to work with you if you want to turn this into PR(s) for a FirebaseRemoteConfigSwift library at

__https://twitter.com/paulbeusterien/status/1448001604536307712?s=20__

First, thanks for your message. To merge SwiftyRemoteConfig into firebase-ios-sdk, I added some tests and found that SwiftyRemoteConfig has some problems for Key-Value Observing and property wrapper. so I cut those features in SwiftyRemoteConfig 0.0.3.

Can I still merge SwiftyRemoteConfig into firebase-ios-sdk without KVO and property wrapper ?

**Comment by paulb777** [comment]
Hi @fumito-ito Thanks for reaching out and nice job with[ SwiftyRemoteConfig](https://github.com/fumito-ito/SwiftyRemoteConfig).

We're definitely interested in exploring merging  SwiftyRemoteConfig into firebase-ios-sdk even without Key-Value Observing and property wrapper.  

Since we connected on Twitter, we've made some more progress.  We now have shared Codable infrastructure across Firebase. See https://github.com/firebase/firebase-ios-sdk/tree/master/FirebaseSharedSwift.  And we've started on a Remote Config implementation that uses it.  See #9084.

So we would need to do a deeper exploration of how to combine the two. Also, we may want to change some of the APIs to be more consistent with Firebase standards.  If you'd like to send a PR, we could merge it into a branch and then work together to get it in shape to merge to our master branch.
